app/src/main/java/com/humberto/tasky/core/presentation/designsystem/components/TaskyCalendarHeader.kt

```diff
@@ -0,0 +1,138 @@
+package com.humberto.tasky.core.presentation.designsystem.components
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.lazy.LazyRow
+import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.lazy.rememberLazyListState
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.humberto.tasky.core.presentation.designsystem.TaskyDarkGray
+import com.humberto.tasky.core.presentation.designsystem.TaskyGray
+import com.humberto.tasky.core.presentation.designsystem.TaskyLight3
+import com.humberto.tasky.core.presentation.designsystem.TaskyOrange
+import com.humberto.tasky.core.presentation.designsystem.TaskyTheme
+import java.time.LocalDate
+
+@Composable
+fun TaskyCalendarHeader(
+    modifier: Modifier = Modifier,
+    onSelectDate: (LocalDate) -> Unit,
+    selectedDate: LocalDate,
+) {
+    val daysList = remember(selectedDate.month, selectedDate.year) {
+        getDaysOfMonth(selectedDate)
+    }
+    val listState = rememberLazyListState()
+
+    var isDayClicked by remember { mutableStateOf(false) }
+    LaunchedEffect(selectedDate) {
+        val selectedDateIndex = daysList.indexOfFirst { it == selectedDate }
+        if (selectedDateIndex != -1 && !isDayClicked) {
+            listState.scrollToItem(selectedDateIndex)
+        }
+        isDayClicked = false
+    }
```

Comment:
Nitpick, but do you really need this `isDayClicked`? I understand that it prevents from clicking somewhere during the animation, but what would be the problem about that? It would cancel the current animation and animate where the user wants to go.

Reply:
Actually the idea here was to only scroll to selected index when selectedDate is coming from initial load(current day), and when selecting a date from the calendar and not when clicking on items from LazyRow/
I renamed this mutable state trying to make it more meaningful .